keanu-python/tests/test_const.py

```diff
@@ -0,0 +1,153 @@
+
+
+def test_const_does_not_take_generic():
+    with pytest.raises(ValueError) as excinfo:
+        kn.Const(Temp())
+
+    assert str(excinfo.value) == "Argument t must be either a numpy array or an instance of numbers.Number. Was given {} instead".format(Temp)
+
+
+def test_const_does_not_take_empty_numpy_tensor():
+    np_tensor = np.array([])
+    with pytest.raises(ValueError) as excinfo:
+        kn.Const(np_tensor)
+
+    assert str(excinfo.value) == "Cannot infer type because tensor is empty"
+
+
+def test_const_takes_numpy_tensor_of_rank_one():
+    np_tensor = np.array([1 ,2])
+    v = kn.Const(np_tensor)
+
+    assert_vertex_value_equal_numpy(v, np_tensor)
+
+
+def test_integer_passed_to_Tensor_creates_ScalarIntegerTensor():
+    t = kn.Tensor(1)
+    assert_java_class(t, "ScalarIntegerTensor")
+    assert t.isScalar()
+
+
+def test_double_passed_to_Tensor_creates_ScalarDoubleTensor():
+    t = kn.Tensor(1.)
+    assert_java_class(t, "ScalarDoubleTensor")
+    assert t.isScalar()
+
+
+def test_bool_passed_to_Tensor_creates_SimpleBooleanTensor():
+    t = kn.Tensor(True)
+    assert_java_class(t, "SimpleBooleanTensor")
+    assert t.isScalar()
+
+
+def test_cannot_pass_generic_to_Tensor():
+    with pytest.raises(ValueError) as excinfo:
+        kn.Tensor(Temp())
+
+    assert str(excinfo.value) == "Generic types in a tensor are not supported. Was given {}".format(Temp)
+
+
+def test_integer_numpy_tensor_passed_to_Tensor_creates_Nd4jIntegerTensor():
+    t = kn.Tensor(np.array([1,2]))
+    assert_java_class(t, "Nd4jIntegerTensor")
+    assert not t.isScalar()
+
+
+def test_double_numpy_tensor_passed_to_Tensor_creates_Nd4jDoubleTensor():
+    t = kn.Tensor(np.array([1.,2.]))
+    assert_java_class(t, "Nd4jDoubleTensor")
+    assert not t.isScalar()
+
+
+def test_bool_numpy_tensor_passed_to_Tensor_creates_SimpleBooleanTensor():
+    t = kn.Tensor(np.array([True, False]))
+    assert_java_class(t, "SimpleBooleanTensor")
+    assert not t.isScalar()
+
+
+def test_cannot_pass_generic_numpy_tensor_to_Tensor():
+    with pytest.raises(ValueError) as excinfo:
+        kn.Tensor(np.array([Temp(), Temp()]))
+
+    assert str(excinfo.value) == "Generic types in a tensor are not supported. Was given {}".format(Temp)
+
+
+def test_cannot_pass_empty_numpy_tensor_to_Tensor():
+    with pytest.raises(ValueError) as excinfo:
+        kn.Tensor(np.array([]))
+
+    assert str(excinfo.value) == "Cannot infer type because tensor is empty"
+
+
+def assert_vertex_value_equal_numpy(v, np_tensor):
+    nd4j_flat = v.getValue().asFlatArray()
+    np_flat = np_tensor.flatten().tolist()
+
+    for i in range(len(np_tensor)):
+        assert nd4j_flat[i] == np_flat[i]
+
+
+def assert_vertex_value_equals_scalar(v, scalar):
+    assert v.getValue().scalar() == scalar
+
+
+def assert_java_class(java_object_wrapper, java_class_str):
+    assert java_object_wrapper.getClass().getSimpleName() == java_class_str
+
+
+class Temp:
```

Comment:
this needs a better name, like `GenericExampleClass`. If you call it `Temp` then it sounds like it should be deleted.